src/synthesizer/particle/stars.py

```diff
@@ -590,29 +839,66 @@ def generate_lnu(
         # Get the integrated spectra in grid units (erg / s / Hz)
         lnu_particle = compute_integrated_sed(*args)
 
-        if parametric_young_stars:
-            return lnu_particle + lnu_parametric
-        else:
-            return lnu_particle
+        return lnu_particle
+
+    def _remove_stars(self, pmask):
+        """
+        Update stars attribute arrays based on a mask, `pmask`.
+
+        Args:
+            pmask (array-like, bool)
+                A boolean mask to remove stars from the object.
+        """
+
+        # Remove the masked stars from this object
+        self.initial_masses = self.initial_masses[~pmask]
+        self.ages = self.ages[~pmask]
+        self.metallicities = self.metallicities[~pmask]
+        if self.masses is not None:
+            self.masses = self.masses[~pmask]
+        if self.coordinates is not None:
+            self.coordinates = self.coordinates[~pmask]
+        if self.tau_v is not None:
+            self.tau_v = self.tau_v[~pmask]
+        if self.alpha_enhancement is not None:
+            self.alpha_enhancement = self.alpha_enhancement[~pmask]
+        if self.velocities is not None:
+            self.velocities = self.velocities[~pmask]
+        if self.current_masses is not None:
+            self.current_masses = self.current_masses[~pmask]
+        if self.s_oxygen is not None:
+            self.s_oxygen = self.s_oxygen[~pmask]
+        if self.s_hydrogen is not None:
+            self.s_hydrogen = self.s_hydrogen[~pmask]
+
+        if self.redshift is not None:
+            if isinstance(self.redshift, np.ndarray):
+                self.redshift = self.redshift[~pmask]
+        if self.smoothing_lengths is not None:
+            if isinstance(self.smoothing_lengths, np.ndarray):
+                self.smoothing_lengths = self.smoothing_lengths[~pmask]
+
+        self.nparticles = len(self.initial_masses)
+        self.nstars = self.nparticles
 
-    def _parametric_young_stars(
+        # Check the arguments we've been given
+        self._check_star_args()
+
+    def parametric_young_stars(
         self,
-        pmask,
         age,
         parametric_sfh,
```

Comment:
Can we not request a class (uninstatiated) here in stead of a string? Feels a bit more intuitive and future proof.